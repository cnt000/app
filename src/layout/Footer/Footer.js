import React from 'react';
import styles from './Footer.module.css';

const Footer = ({ copyright, privacy, contact, company, companyAddress, disclaimer }) => (
  <footer className={styles.footer}>
    <p>
      <span className={styles.copyright}>{copyright}</span> -
      <a className={styles.privacy} href="http://www.pungilandia.com">
        Privacy
      </a>{' '}
      -
      <a className={styles.contactus} href="mailto:info@pungilandia.com">
        Contatti
      </a>
    </p>
    <address>
      Azienda Agricola Pungilandia Rocchi Domenico
      <br />
      Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B
      Iscrizione REA Pescara Nr.97422
    </address>
    <p className={styles.laws}>
      Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse
      Negli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9
      Dicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001
      VAB
    </p>
  </footer>
);

export default Footer;
