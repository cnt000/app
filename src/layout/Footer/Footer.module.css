.footer {
  grid-area: footer;
  padding: 0.5rem;
  text-align: center;
}

.contactus {
  font-size: 16px;
  margin-top: 0.4em;
}

.address,
.laws {
  font-size: 12px;
  margin-top: 0.4rem;
  line-height: 1.2;
}

.laws {
  width: 80%;
  margin: 0 auto;
  font-weight: 300;
}

.laws {
  display: none;
}


@media only screen and (min-width: 1025px) {
  .laws {
    display: block;
    margin-top: 0.4rem;
    width: 60%;
  }
}
