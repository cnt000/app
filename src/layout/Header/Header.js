import React from 'react';
import logo from '../../images/logo.png';
import styles from './Header.module.css';

const Header = ({ openMenuLabel, title, titleDescription, goToCartLabel, setMenuOpen }) => {
  return (
    <header className={styles.header}>
      <div>
        <button className={styles.burger} onClick={setMenuOpen}>{openMenuLabel}</button>
      </div>
      <div className={styles.content}>
        <div className={styles.logo}>
          <img src={logo} alt="pungilandia logo" />
        </div>
        <div className={styles.title}>
          <h1>{title}</h1>
          <p>{titleDescription}</p>
        </div>
      </div>
      <div>
        <button className={styles.cart}>{goToCartLabel}</button>
      </div>
    </header>
  );};

export default Header;

