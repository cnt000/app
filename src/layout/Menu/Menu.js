import React from 'react';
import styles from './Menu.module.css';

const Menu = ({ isOpen, setMenuOpen }) => (
  <>
    <nav className={`${styles.menu} ${isOpen ? styles.open : ''}`}>
      <div className={styles.header}>
        <button className={styles.close} onClick={(setMenuOpen)}>
          chiudi il menu
        </button>
      </div>
      <ul className={styles.entries}>
        <li>
          <a href="https://www.pungilandia.com">Chi siamo</a>
        </li>
        <li>
          <a href="https://www.pungilandia.com">Lista piante</a>
        </li>
        <li>
          <a href="https://www.pungilandia.com">Novit√†</a>
        </li>
        <li>
          <a href="https://www.pungilandia.com">Prodotti speciali</a>
        </li>
        <li>
          <a href="https://www.pungilandia.com">Consigli per coltivare</a>
        </li>
        <li>
          <a href="https://www.pungilandia.com">Ordini</a>
        </li>
        <li>
          <a href="https://www.pungilandia.com">Foto-Show</a>
        </li>
        <li>
          <a href="https://www.pungilandia.com">E-mail</a>
        </li>
      </ul>
    </nav>
    <div className={styles.overlay}></div>
  </>
);

export default Menu;
