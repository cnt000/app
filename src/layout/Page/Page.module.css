.pageLayout {
  position: relative;
  height: 100vh;
  display: grid;
  grid-template-columns: 9rem 1fr;
  grid-template-rows: 3.2rem 1fr 5rem;
  grid-template-areas:
    'header  header'
    'content  content'
    'footer  footer';
}

@media only screen and (min-width: 1025px) {
  body {
    grid-template-rows: 9rem 1fr 7rem;
    grid-template-columns: 12rem 1fr;
    grid-template-areas:
      'sidebar  header'
      'sidebar  content'
      'sidebar  footer';
  }
}
