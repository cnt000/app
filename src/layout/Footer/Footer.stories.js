import React from 'react';

import Footer from './Footer';
import { withLabels } from '../../features/Labels/withLabels';

export default {
  component: Footer,
  title: 'Footer',
};

const footer = () => <Footer />;

export const Default = withLabels(footer);
