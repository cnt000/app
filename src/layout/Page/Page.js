import React, { useState } from 'react';

import Header from '../Header/Header';
import Menu from '../Menu/Menu';
import Footer from '../Footer/Footer';
import styles from './Page.module.css';

const Page = (props) => {
  const [menuOpen, setMenuOpen] = useState(false);
  return (
    <div className={styles.pageLayout}>
      <Header {...props} setMenuOpen={() => setMenuOpen(true)} />
      <Menu isOpen={menuOpen} setMenuOpen={() => setMenuOpen(false)} Link={props.Link}/>
      {props.children}
      <Footer {...props} />
    </div>
  );
};

export default Page;
