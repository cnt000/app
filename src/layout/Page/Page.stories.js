import React from 'react';
import { BrowserRouter as Router, Link } from 'react-router-dom';

import Page from './Page';
import Plp from '../Plp/Plp';
import Pdp from '../Pdp/Pdp';
import Article from '../Article/Article';
import Home from '../Home/Home';

import labels from '../../data/labels/labels.json';
import LabelsContext from '../../features/Labels';
import product from '../../data/products/clavifolius.json';
import articleContent from '../../data/articles/consigliColtivazione.md';

export default {
  component: Page,
  title: 'Page',
};

const plp = <Plp {...product} />;
const pdp = <Pdp {...product} />;
const article = <Article articleContent={articleContent} />;
const home = <Home />;

export const PagePlp = () => (
  <LabelsContext.Provider value={labels}>
    <Router>
      <Page>{plp}</Page>
    </Router>
  </LabelsContext.Provider>
);

export const PagePdp = () => (
  <LabelsContext.Provider value={labels}>
    <Router>
      <Page {...labels}>
        {pdp}
      </Page>
    </Router>
  </LabelsContext.Provider>
);

export const PageArticle = () => (
  <LabelsContext.Provider value={labels}>
    <Router>
      <Page {...labels}>
        {article}
      </Page>
    </Router>
  </LabelsContext.Provider>
);

export const PageHome = () => (
  <LabelsContext.Provider value={labels}>
    <Router>
      <Page {...labels}>
        {home}
      </Page>
    </Router>
  </LabelsContext.Provider>
);
