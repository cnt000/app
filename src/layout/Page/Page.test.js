import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import { BrowserRouter as Router, Link } from 'react-router-dom';

import Page from './Page';
import Header from '../Header/Header';
import Footer from '../Footer/Footer';
import Menu from '../Menu/Menu';
import labels from '../../data/labels/labels.json';
import LabelsContext from '../../features/Labels';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const tree = renderer
    .create(
      <LabelsContext.Provider value={labels}>
        <Router>
          <Page>{<p>test</p>}</Page>
        </Router>
      </LabelsContext.Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Click without enzyme', () => {
  let container;
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation((init) => [init, setState]);

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
    jest.clearAllMocks();
  });

  it('render Header, Menu, Footer', () => {
    const wrapper = shallow(
      <Page {...labels}>
        <span>abc</span>
      </Page>,
    );
    expect(wrapper.find(Header).length).toBe(1);
    expect(wrapper.find(Menu).length).toBe(1);
    expect(wrapper.find(Footer).length).toBe(1);
  });

  it('render Header with setMenuOpen prop', () => {
    const wrapper = shallow(
      <Page {...labels}>
        <span>abc</span>
      </Page>,
    );
    expect(wrapper.find(Header).length).toBe(1);
    expect(wrapper.find(Header).props().setMenuOpen).toEqual(
      expect.any(Function),
    );
  });

  it('render Menu with setMenuOpen prop', () => {
    const wrapper = shallow(
      <Page {...labels}>
        <span>abc</span>
      </Page>,
    );
    expect(wrapper.find(Menu).length).toBe(1);
    expect(wrapper.find(Menu).props().setMenuOpen).toEqual(
      expect.any(Function),
    );
    expect(wrapper.find(Menu).props().isOpen).toEqual(expect.any(Boolean));
  });
});
