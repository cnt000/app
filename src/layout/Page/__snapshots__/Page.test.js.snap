// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="pageLayout"
>
  <header
    className="header"
  >
    <div>
      <button
        className="burger"
        onClick={[Function]}
      >
        apri il menu
      </button>
    </div>
    <div
      className="content"
    >
      <div
        className="logo"
      >
        <img
          alt="pungilandia logo"
          src="logo.png"
        />
      </div>
      <div
        className="title"
      >
        <h1>
          Pungilandia
        </h1>
        <p>
          Piante grasse, strane ed insolite
        </p>
      </div>
    </div>
    <div>
      <button
        className="cart"
      >
        vai al carrello
      </button>
    </div>
  </header>
  <nav
    className="menu "
  >
    <div
      className="header"
    >
      <button
        className="close"
        onClick={[Function]}
      >
        chiudi il menu
      </button>
    </div>
    <ul
      className="entries"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Chi siamo
        </a>
      </li>
      <li>
        <a
          href="/plp"
          onClick={[Function]}
        >
          Lista piante
        </a>
      </li>
      <li>
        <a
          href="/pdp"
          onClick={[Function]}
        >
          Novità
        </a>
      </li>
      <li>
        <a
          href="/article"
          onClick={[Function]}
        >
          Prodotti speciali
        </a>
      </li>
      <li>
        <a
          href="/test2"
          onClick={[Function]}
        >
          Consigli per coltivare
        </a>
      </li>
      <li>
        <a
          href="/test3"
          onClick={[Function]}
        >
          Ordini
        </a>
      </li>
      <li>
        <a
          href="/test4"
          onClick={[Function]}
        >
          Foto-Show
        </a>
      </li>
      <li>
        <a
          href="/test5"
          onClick={[Function]}
        >
          E-mail
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="overlay"
  />
  <p>
    test
  </p>
  <footer
    className="footer"
  >
    <p>
      <span
        className="copyright"
      >
        Copyright © 2020
      </span>
       -
       
      <a
        className="privacy"
        href="http://www.pungilandia.com"
      >
        privacy
      </a>
       
      -
       
      <a
        className="contactus"
        href="mailto:info@pungilandia.com"
      >
        contatti
      </a>
    </p>
    <address>
      Azienda Agricola Pungilandia Rocchi Domenico
      <br />
      Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B
Iscrizione REA Pescara Nr.97422
    </address>
    <p
      className="laws"
    >
      Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse
Negli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9
Dicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001
VAB
    </p>
  </footer>
</div>
`;
